hazelcast/src/main/java/com/hazelcast/nio/tcp/WriteHandler.java

```diff
@@ -172,7 +172,7 @@ private void unschedule() {
         // So the outputBuffer is empty, so we are going to unschedule ourselves.
         scheduled.set(false);
 
-        if (writeQueue.isEmpty() && urgentWriteQueue.isEmpty()) {
+        if (writeQueue.isEmpty() && urgentWriteQueue.isEmpty() && currentPacket == null) {
```

Comment:
@sancar I'm not understanding the point you try to make.

@general
We should have a a look at cleaning up the IO system. But probably best to make the whole thing pluggable so we can keep using the old and swap in the new if needed. This way we can fall back upon something battle hardened.